crates/sequencer/src/db_provider/mod.rs

```diff
@@ -32,27 +33,23 @@ impl<C: sov_modules_api::Context> DbProvider<C> {
         self.evm.get_chain_config(&mut working_set)
     }
 
-    pub fn last_block_tx_hashes(&self) -> Vec<B256> {
+    pub fn last_block_tx_hashes(&self) -> RpcResult<Vec<B256>> {
         let mut working_set = WorkingSet::<C>::new(self.storage.clone());
-        let rich_block = self
-            .evm
-            .get_block_by_number(None, None, &mut working_set)
-            .unwrap()
-            .unwrap();
-        match rich_block.inner.transactions {
-            BlockTransactions::Hashes(hashes) => hashes,
-            _ => vec![],
+        let rich_block = self.evm.get_block_by_number(None, None, &mut working_set)?;
+        let hashes = rich_block.map(|b| b.inner.transactions);
```

Comment:
No need to map if you have match over Some/None later. So either map or match IMO

Reply:
- `rich_block` before map is `Option<Rich<Block>>`. 
- `hashes` becomes after map `Option<BlockTransactions>`
- The match checks that we only return hashes and an empty vector otherwise (for `None` and for `BlockTransaction` other variants)